kubernetes/client/ws_client_test.py

```diff
@@ -32,6 +36,24 @@ def test_websocket_client(self):
                 ]:
             self.assertEqual(get_websocket_url(url), ws_url)
 
+    @patch.object(websocket._http, '_can_use_sni')
+    @patch.object(websocket._http, '_open_socket')
+    @patch.object(ssl.SSLContext, 'load_verify_locations')
+    def test_use_of_correct_ca_cert(self, load_verify_locations, *ignored):
+        # arrange
+        expected = '/expected-ca.pem'
+        configuration.ssl_ca_cert = expected
+        load_verify_locations.side_effect = Exception("need not go further")
+
+        # act
+        with self.assertRaises(Exception):
```

Comment:
It's just a trick to unwind the stack (force an early return deeper in the call graph). 

The SSL bundle loading in python happens deep within the websocket library, and I really _didn't_ want to test that code's implementation, but could think of no other way to assert that the SSL configuration made in the kube client gets propagated properly to python-ssl.

That propagation is at the heart of this bug, and testing just the interface between the kube client and websocket library would not provide any increased certainty, and would not reveal this bug since the bug is an _interface mismatch_ between this project and its immediate dependency. I.e. a breaking change in the dependent project.

Since the python SSL library is well documented and understood, testing what data eventually propagates to _it_ should increase confidence that the intention underpinning dependent functions invocations from this project yield the expected effect.

The other option would have been to simply bump the requirements spec to websocket-client >= 0.41.0 and just change the configuration key (one character) in the code, but I felt uncertain of what implications bumping lower bounds for the package requirement would entail. Some teams support a wide range of package versions for various reasons, while other projects pin versions hard (to exact revisions). Since this project uses version ranges I assumed there was probably a good reason for that, a reason I however am unaware of. Thus sticking with existing requirements and compensating for change in interface by conditional branching.

Does that answer the question sufficiently, and provide enough context to see the reasoning?